### PR TITLE
Halo ID url params, RTD dupe protection, doc update

### PR DESCRIPTION
**Pull request by anthonylauzon**
## Type of change
- [X] Bugfix

This is a small update to the Halo code.  It adds configurable URL parameters to the ID module, adds array-level data duplication protection to the RTD module, and updates the RTD documentation.

- contact email of the adapter’s maintainer
prebid@audigent.com

## Other information
Halo RTD Module: Enable Per-Bidder Data - https://github.com/prebid/Prebid.js/pull/7060
Halo RTD Module: FPD 2.0 Updates & add ID system tests  - https://github.com/prebid/Prebid.js/pull/6505

### CODE REVIEW

**Comment by anthonylauzon** [comment]
@ChrisHuie this is similar to the last one, a few minor changes to the id & rtd system

File: modules/haloIdSystem.js [hunk 1]
```diff
@@ -42,7 +56,12 @@ export const haloIdSubmodule = {
    * @returns {IdResponse|undefined}
    */
   getId(config) {
-    const url = `https://id.halo.ad.gt/api/v1/pbhid`;
+    if (!utils.isPlainObject(config.params)) {
+      config.params = {};
+    }
+    const url = paramOrDefault(config.params.url,
+      `https://id.halo.ad.gt/api/v1/pbhid`,
+      config.params.urlArg);
```
Comment (smenzer) [inline modules/haloIdSystem.js:64]: this parameter isn't documented in your .md file...should it be? what do you expect here?

**Review by smenzer — CHANGES_REQUESTED** [review]
can you please also submit a prebid docs PR for the new `url` config param?

**Comment by anthonylauzon** [comment]
@smenzer update docs for url arg and there is a pr for the website here: https://github.com/prebid/prebid.github.io/pull/3154

**Review by smenzer — APPROVED** [review]
LGTM

**Comment by MartianTribe** [comment]
https://github.com/prebid/prebid.github.io/pull/3154

**Pull request merged**